frame/evm/src/runner/stack.rs

```diff
@@ -200,7 +200,15 @@ where
 		};
 
 		let metadata = StackSubstateMetadata::new(gas_limit, config);
-		let state = SubstrateStackState::new(&vicinity, metadata);
+		let weight_info = T::GasWeightMapping::gas_to_weight(gas_limit, true);
+		// Used to record the external costs in the evm through the StackState implementation
+		let weight_info = WeightInfo {
+			ref_time_limit: weight_info.ref_time(),
+			proof_size_limit: weight_info.proof_size(),
```

Comment:
This code assume that the `proof_size_limit` is derived from the `gas_limit`, but this is not the case.

Firstly, the calculation of the proof_size_limit must take into account the encoded size of the transaction, and secondly, in some cases it is the user who will provide it directly. 

This is why the proof_size_limit should not be computed by frontier but be an optional parameter of each pallet call